packages/core/RNSentryAndroidTester/app/src/androidTest/java/io/sentry/rnsentryandroidtester/RNSentryMapConverterTest.kt

```diff
@@ -359,4 +363,40 @@ class MapConverterTest {
 
         assertEquals(actual, expectedMap1)
     }
+
+    @Test
+    fun testJsonObjectToReadableMap() {
+        val json =
+            JSONObject().apply {
+                put("stringKey", "stringValue")
+                put("booleanKey", true)
+                put("intKey", 123)
+            }
+
+        val result = RNSentryMapConverter.jsonObjectToReadableMap(json)
+
+        assertNotNull(result)
+        assertTrue(result is JavaOnlyMap)
+        assertEquals("stringValue", result.getString("stringKey"))
+        assertEquals(true, result.getBoolean("booleanKey"))
+        assertEquals(123, result.getInt("intKey"))
+    }
+
+    @Test
+    fun testMapToReadableMap() {
+        val map =
+            mapOf(
+                "stringKey" to "stringValue",
+                "booleanKey" to true,
+                "intKey" to 123,
+            )
+
+        val result = RNSentryMapConverter.mapToReadableMap(map)
+
+        assertNotNull(result)
+        assertTrue(result is JavaOnlyMap)
+        assertEquals("stringValue", result.getString("stringKey"))
+        assertEquals(true, result.getBoolean("booleanKey"))
+        assertEquals(123, result.getInt("intKey"))
+    }
```

Comment:
We should also ensure nested maps and arrays are handled because some Sentry init options require those types.

Reply:
Makes sense 👍 

The nested maps/arrays should be handled with the https://github.com/getsentry/sentry-react-native/pull/4469 enhancement/refactoring discussed above https://github.com/getsentry/sentry-react-native/pull/4451#discussion_r1923360541

I'll also extend the json handling to support nested structures with https://github.com/getsentry/sentry-react-native/pull/4470